### PR TITLE
Features/Add directory header #162561699

### PR DESCRIPTION
**Pull request by jrubenoff**
[Pivotal link.](https://www.pivotaltracker.com/story/show/162561700)

### Changes

- Add partials for...
   - Rent directory header
   - Buy directory header
   - Rent callout
   - Buy callout
- Use rental partials on "Browse Listings" page
- Add an ID to the listings section, letting us smooth-scroll to it from the Buy directory header
- Update results header styles to match new headers
- Update "sign up for email" block styles on homepage

### To-Do

- [x] Account for all directory page states (Matching listings, no listings, etc.)
- [x] Browser / multi-language testing
- [x] Write review instructions

### CODE REVIEW

File: app/assets/json/translations/locale-en.json [hunk 1]
```diff
@@ -627,6 +627,42 @@
         "DUE_TO_HIGH_CALL_VOLUME": "Due to high call volume you may hear a message.",
         "OFFICE_HOURS": "Office Hours"
     },
+    "LISTINGS_FOR_RENT": {
```
Comment (jrubenoff) [inline app/assets/json/translations/locale-en.json]: @chrisdolendo I tried running `grunt translations` to keep everything nice and alphabetical, but I ended up with a pretty large diff beyond what's in this PR, so I kept it as-is.

Would it make sense to run `grunt-translations` on a branch off master in a separate story?

**Comment by akegan** [comment]
based on https://github.com/Exygy/sf-dahlia-web/pull/1104 it looks like we will probably actually use the same listings.html for rentals and ownership, which should make things pretty easy to wire up!

**Comment by akegan** [comment]
To make it clear when we're going to actually implement the ownership directory header, I've created a follow-up ticket https://www.pivotaltracker.com/story/show/163633411

**Comment by callaghanc** [comment]
i've reached out to @jrubenoff over slack to sync up on this PR so i can help out on it

**Comment by akegan** [comment]
@callaghanc looks good to me so far, we'll just have to be careful about the isOwnership -> isSale transition to make sure we don't introduce any sneaky bugs anywhere

**Comment by callaghanc** [comment]
ok i've added/updated basic e2e tests for using the homepage Rent and Buy links to go to the appropriate listings pages. ideally we'd have some testing around the content of those pages, but i believe that will have to come a little later in a future listings-page-related story after we've established the view testing that we're working on in other PRs. @jrubenoff i think that concludes my part on this for now, so will leave it to you to complete this PR.

**Comment by jrubenoff** [comment]
> we don't think there is one set of terms that fits perfectly in every situation

Your conclusions make sense! FWIW, I've come to think of it as:

- "Sale units" and "rental units" are the nouns we use
- "Buy" and "Rent" are actions users can take
- "Ownership" is more of an internal MOHCD term than a user-facing term, so we should avoid using it in the app

**Comment by jrubenoff** [comment]
Alrighty, this is ready for review.

**Comment by exygy-dev** [comment]
@jrubenoff That breakdown makes sense. @akegan said something quite similar in our discussion so it looks like we have a shared understanding of these terms now.

**Comment by akegan** [comment]
ok! I'll review do a final code review ASAP

**Comment by akegan** [comment]
looks like both call-outs are showing up on the rentals page? 
image

**Comment by callaghanc** [comment]
@akegan I'll take a look

**Comment by callaghanc** [comment]
@akegan hmm i don't see it, is https://dahlia-full-pr-1108.herokuapp.com/listings/for-rent the page where you're seeing this?

**Comment by akegan** [comment]
Looks like nav bar titles weren't updated for mobile view (issue with rent/buy nav bar PR that was already merged) @callaghanc do you want to include that in this PR or open up a ticket?
image

**Comment by callaghanc** [comment]
@akegan re: mobile view, is there design guidance on which icons to use for the new Rent and Buy links in the mobile nav? if yes, let's add that to this PR. if no, let's not have this PR held back waiting for guidance on that.

**Comment by callaghanc** [comment]
cc @jrubenoff see my comment above - do you have guidance for what icons to use in the mobile nav for the new Rent and Buy links?

**Comment by jrubenoff** [comment]
Let's remove the icons. (Opened [#163772973](https://www.pivotaltracker.com/story/show/163772973) for this a few hours ago.)

File: app/assets/javascripts/listings/components/browse-listings.html.slim [hunk 2]
```diff
@@ -92,6 +85,8 @@ section.results-section ng-if="!$ctrl.parent.openListings.length"
       .large-12.columns
         property-card listing='listing' ng-repeat="listing in $ctrl.parent.openListings"
 
+ng-include src="'listings/templates/_rental-callout.html'" ng-if="$ctrl.tenureType == 'rental' && !$ctrl.parent.hasEligibilityFilters() || $ctrl.openMatchListings.length"
```
Comment (akegan) [inline app/assets/javascripts/listings/components/browse-listings.html.slim]: @callaghanc  seems like the double callback is coming from here. I'm not sure what the goal of the logic is, but maybe it should be $ctrl.tenureType == 'rental' && (!$ctrl.parent.hasEligibilityFilters() || $ctrl.openMatchListings.length) ?

Comment (callaghanc) [reply app/assets/javascripts/listings/components/browse-listings.html.slim]: yes that makes sense, i'll push that update

**Comment by callaghanc** [comment]
thanks for the guidance on the mobile nav icons @jrubenoff, i can remove them and add the Rent and Buy links in this PR, if you don't mind us doing it here instead of in the story that you linked

**Comment by callaghanc** [comment]
@akegan @jrubenoff I've pushed these updates to this pr:
1) Updated condition on listings page callouts so that correct callout appears on sale vs rental pages when eligibility is selected.
2) Removed the icons from the mobile nav items.
3) Added the Rent and Buy links to the mobile nav when the SHOW_SALE_LISTINGS env var is set.

**Comment by akegan** [comment]
@jrubenoff did you want to do a final QA before merge or are we good to go?

**Comment by jrubenoff** [comment]
Yup, we're good to merge

**Pull request merged**